### PR TITLE
feat(sheet): support rule registry in auto fill

### PR DESCRIPTION
**Pull request by yuhongz**

### CODE REVIEW

**Comment by Dushusir** [comment]
excellent code

File: packages/ui-plugin-sheets/src/services/auto-fill/auto-fill.service.ts [hunk 1]
```diff
@@ -91,11 +91,21 @@ export class AutoFillService extends Disposable implements IAutoFillService {
...
-        ];
-        // this._applyType = APPLY_TYPE.SERIES;
+        ].sort((a, b) => b.priority - a.priority);
         this._isFillingStyle = true;
     }
 
+    registryRule(rule: IAutoFillRule) {
+        // if rule.type is used, console error
+        if (this._rules.find((r) => r.type === rule.type)) {
+            console.error(`[ERROR]: registry rule failed, type '${rule.type}' already exist!`);
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/services/auto-fill/auto-fill.service.ts:101]: Throw an error?

File: packages/ui-plugin-sheets/src/services/auto-fill/type.ts [hunk 2]
```diff
@@ -23,14 +23,16 @@ export interface ICopyDataInType {
 export type ICopyDataInTypeIndexInfo = number[];
 
 export interface IAutoFillRule {
-    type: DATA_TYPE;
+    type: string;
     match: (cellData: Nullable<ICellData>) => boolean;
     isContinue: (prev: IRuleConfirmedData, cur: Nullable<ICellData>) => boolean;
     applyFunctions?: APPLY_FUNCTIONS;
+    // 优先级
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/services/auto-fill/type.ts:30]: Write comments in English.

File: packages/ui-plugin-sheets/src/services/auto-fill/auto-fill.service.ts (hunk 1 above)
Comment (yuhongz) [reply packages/ui-plugin-sheets/src/services/auto-fill/auto-fill.service.ts:101]: good idea

**Review by wzhudev — APPROVED** [review]

**Pull request merged**